subsys/usb/class/cdc_acm.c

```diff
@@ -919,6 +929,22 @@ static int cdc_acm_line_ctrl_get(struct device *dev,
 	return -ENOTSUP;
 }
 
+static int cdc_acm_line_ctrl_callback_set(struct device *dev,
+					  uart_line_ctrl_callback_t callback,
+					  void *user_data)
+{
+	struct cdc_acm_dev_data_t * const dev_data = DEV_DATA(dev);
+
+	if (dev_data->line_ctrl_cb != NULL) {
+		return -EINVAL;
+	}
+
+	dev_data->line_ctrl_cb = callback;
+	dev_data->line_ctrl_cb_data = user_data;
```

Comment:
This looks a little odd to me: you can pass a null callback, but only if no callback has been set.  Once a callback is set, you can't clear it.

If the intent that this function can be invoked only once, then that should be documented.  If it's possible to set the callback multiple times then a lock must be provided to ensure that use of the callback is consistent with a configured pair of callback and userdata.

Without knowing much about this I think the simplest solution is document that the function can be successfully invoked only once, and that passing a null callback returns `-EINVAL` while passing a non-null callback if one has already been set returns `-EBUSY` or `-EALREADY`.  `user_data` can be null or not, but you can't change it once the callback has been set.

If that doesn't satisfy the use case document and implement something that does.

Reply:
Hi Peter.  I had a look at the other callback implementations and you're right - any callback set will overwrite any previous callback set.

Changed this to match.